From: Heiko Becker <heirecka@exherbo.org>
Date: Thu, 24 Sep 2015 20:54:41 +0000
Subject: qca-ossl: Fix build without SSLv3
X-Git-Url: http://quickgit.kde.org/?p=qca.git&a=commitdiff&h=20a587d77636186edb044cd2b71d6d90fe98d232
---
qca-ossl: Fix build without SSLv3

This fixes building with LibreSSL >= 2.3.0 which has removed support
for SSLv3 completely. As far as I know OpenSSL can be configured to
build without it, so it might be helpful there as well.

REVIEW: 125386
---


--- a/plugins/qca-ossl/qca-ossl.cpp
+++ b/plugins/qca-ossl/qca-ossl.cpp
@@ -5403,9 +5403,11 @@
 			ctx = SSL_CTX_new(SSLv2_client_method());
 			break;
 #endif
+#ifndef OPENSSL_NO_SSL3_METHOD
 		case TLS::SSL_v3:
 			ctx = SSL_CTX_new(SSLv3_client_method());
 			break;
+#endif
 		case TLS::TLS_v1:
 			ctx = SSL_CTX_new(TLSv1_client_method());
 			break;
